pkg/controller/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -695,19 +713,31 @@ func (r *ReconcileAWSPrivateLink) reconcileVPCEndpointService(awsClient *awsClie
 	for _, allowed := range permResp.AllowedPrincipals {
 		oldPerms.Insert(aws.StringValue(allowed.Principal))
 	}
-	desriredPerms := sets.NewString(aws.StringValue(stsResp.Arn))
+	// desiredPerms is the set of Allowed Principals that will be configured for the cluster's VPC Endpoint Service.
+	// desiredPerms only contains the IAM entity used by Hive (defaultARN) by default but may contain additional Allowed Principal
+	// ARNs as configured within cd.Spec.Platform.AWS.PrivateLink.AdditionalAllowedPrincipals.
+	defaultARN := aws.StringValue(stsResp.Arn)
+	desiredPerms := sets.NewString(defaultARN)
+	if allowedPrincipals := cd.Spec.Platform.AWS.PrivateLink.AdditionalAllowedPrincipals; allowedPrincipals != nil {
+		for _, allowedPrincipal := range *allowedPrincipals {
+			desiredPerms.Insert(allowedPrincipal)
+		}
+	}
 
-	if !desriredPerms.Equal(oldPerms) {
+	if !desiredPerms.Equal(oldPerms) {
 		modified = true
 		input := &ec2.ModifyVpcEndpointServicePermissionsInput{
 			ServiceId: serviceConfig.ServiceId,
 		}
-		if added := desriredPerms.Difference(oldPerms).List(); len(added) > 0 {
+		if added := desiredPerms.Difference(oldPerms).List(); len(added) > 0 {
 			input.AddAllowedPrincipals = aws.StringSlice(added)
 		}
-		if removed := oldPerms.Difference(desriredPerms).List(); len(removed) > 0 {
+		if removed := oldPerms.Difference(desiredPerms).List(); len(removed) > 0 {
 			input.RemoveAllowedPrincipals = aws.StringSlice(removed)
 		}
```

Comment:
nit: you could save a negligible amount of processing/memory by waiting until inside the conditions to do `.List()`.